client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,155 @@
+            if (recipient) {
+                postZoop({
+                    content: content,
+                    authorId: currentUser?.user.id,
+                    receiverId: recipient.id
+                })
+                .unwrap()
+                .then((payload) => {
+                    console.log('fulfilled', payload);
+                    setLastZoopSentId(payload.zoop.id)
+                })
+                .catch((error) => console.error('rejected', error));
+              
+                setZoopSent(true)
+            } else {
+                throw new Error('No such user exists. Please select a different username');
+                // TODO: How best to render this error in UI?
+            }
+            
+        }
+        
+    }
+
+    return ( //TODO: MUI suggests adding a wrapper element so that Tooltip works when Fab is disabled
+        <>
+            <Tooltip 
+                title="Send a Zoop"
+            >
+                <Fab
+                    disabled={!token}
+                    sx={{
+                        position: "absolute",
+                        bottom: 25,
+                        right: 25
+                    }}
+                    onClick={() => setOpen(true)}
+                > <SendIcon />
+                </Fab>
+            </Tooltip>
+
+            {/* Render a success dialog if Zoop was sent and successfully integrated into DB,
+                or an input dialog if Zoop has not yet been sent */}
+            {zoopSent ? (
+                <Dialog
+                    open={open}
+                    onClose={() => setOpen(false)}
+                >
+                    <DialogTitle>Zoop Sent!</DialogTitle>
+                    <DialogContent>
+                        <DialogContentText> 
+                            <Link
+                                component="button"
+                                onClick={() => navigate(`/zoops/${lastZoopSentId}`)}
+                            > 
+                                Go to this page to view your Zoop.
+                            </Link>
+                        </DialogContentText>
+
+                    </DialogContent>
+                </Dialog>
+            ) : (
+                <Dialog
+                    open={open}
+                    onClose={() => setOpen(false)}
+                >
+                    <DialogTitle>Compose a Zoop</DialogTitle>
+                    <DialogContent>
+                        <TextField
+                            autoFocus
+                            required
+                            margin="dense"
+                            label="Recipient"
+                            type="text"
+                            placeholder="Username"
+                            onChange={(e) => setUsername(e.target.value)}
+                            value={username}    
+                        />
```

Comment:
I would not suggest a `TextField` here - I would recommend some sort of typeahead/dropdown selection. Check out `Autocomplete` in MUI: https://mui.com/material-ui/react-autocomplete

In that case, you would use map over the users to create an array of objects, where the label is the username, and the ID is the user's ID.

You could add a Wish List task for enabling search as you type to make it more performant (this would work better if there were a lot of users): https://mui.com/material-ui/react-autocomplete/#search-as-you-type

Reply:
@dyazdani this comment got marked as outdated, want to make sure you saw it

Reply:
Yes, I just finished this!